Lib/test/test_types.py

```diff
@@ -703,22 +703,62 @@ def test_hash(self):
         self.assertEqual(hash(int | str), hash(str | int))
         self.assertEqual(hash(int | str), hash(typing.Union[int, str]))
 
-    def test_instancecheck(self):
-        x = int | str
-        self.assertIsInstance(1, x)
-        self.assertIsInstance(True, x)
-        self.assertIsInstance('a', x)
-        self.assertNotIsInstance(None, x)
-        self.assertTrue(issubclass(int, x))
-        self.assertTrue(issubclass(bool, x))
-        self.assertTrue(issubclass(str, x))
-        self.assertFalse(issubclass(type(None), x))
-        x = int | None
-        self.assertIsInstance(None, x)
-        self.assertTrue(issubclass(type(None), x))
-        x = int | collections.abc.Mapping
-        self.assertIsInstance({}, x)
-        self.assertTrue(issubclass(dict, x))
+    def test_instancecheck_and_subclasscheck(self):
+        for x in [int | str, typing.Union[int, str]]:
```

Comment:
Minor style thing;

```suggestion
        for x in (int | str, typing.Union[int, str]):
```

Tuple parentheses are interpreted as grouping things, compared to a specific list object (+ avoid runs of brackets)

Reply:
Good point! `()` and `[]` surely have a different meaning here. Thanks, I will change them tomorrow.

Reply:
FWIW, I disagree with @merwok -- anything you iterate over surely is more like a list than like a tuple.

Reply:
(But I also don't want to hold up the PR, so I'll merge regardless.)